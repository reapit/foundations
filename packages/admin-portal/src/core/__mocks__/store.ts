export default {
  dispatch: jest.fn(),
  state: {},
}
