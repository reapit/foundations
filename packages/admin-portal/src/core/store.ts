import { createStore, applyMiddleware, compose, combineReducers, Store as ReduxStore, Dispatch } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { fork, all } from '@redux-saga/core/effects'
import { ReduxState } from '../types/core'
import appDetail from '@/reducers/app-detail'
import error from '@/reducers/error'
import approvals from '@/reducers/approvals'
import devsManagement from '@/reducers/devs-management'
import developerSetStatus from '@/reducers/developer-set-status'
import revisionDetail from '@/reducers/revision-detail'
import appDeleteReducer from '@/reducers/app-delete'
import adminApps from '@/reducers/apps-management'
import statisticsReducer from '@/reducers/statistics'
import noticationMessage from '@/reducers/notification-message'

import { appDetailSagas, revisionDetailSagas, appDeleteSagas, appsManagementSagas } from '@/sagas/apps'
import { statisticsSagas } from '@/sagas/statistics'
import { approvalsSagas } from '@/sagas/approvals'
import { devsManagementSagas, developerSetStatusSagas } from '@/sagas/developers'

import { injectSwitchModeToWindow } from '@reapit/elements'

export class Store {
  static _instance: Store

  static get instance() {
    if (!Store._instance) {
      Store._instance = new Store()
    }

    return Store._instance
  }

  static isProd = process.env.NODE_ENV === 'production'

  static sagaMiddleware = createSagaMiddleware()

  static reducers = combineReducers<ReduxState>({
    appDetail,
    error,
    appsManagement: adminApps,
    approvals,
    devsManagement,
    developerSetStatus,
    revisionDetail,
    appDelete: appDeleteReducer,
    noticationMessage,
    statistics: statisticsReducer,
  })

  static sagas = function*() {
    yield all([
      fork(appDetailSagas),
      fork(approvalsSagas),
      fork(devsManagementSagas),
      fork(developerSetStatusSagas),
      fork(revisionDetailSagas),
      fork(appDeleteSagas),
      fork(appsManagementSagas),
      fork(statisticsSagas),
    ])
  }

  static composeEnhancers =
    !Store.isProd && window && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose

  reduxStore: ReduxStore<ReduxState>

  constructor() {
    injectSwitchModeToWindow()
    const composed = Store.composeEnhancers(applyMiddleware(Store.sagaMiddleware))

    this.reduxStore = createStore(Store.reducers, composed)

    Store.sagaMiddleware.run(Store.sagas)

    this.hotModuleReloading()
  }

  hotModuleReloading() {
    const hotModule = (module as any).hot

    if (hotModule) {
      // Enable Webpack hot module replacement for reducers
      hotModule.accept('../reducers', () => {
        this.reduxStore.replaceReducer(Store.reducers)
      })
    }
  }

  get dispatch(): Dispatch {
    return this.reduxStore.dispatch
  }

  get state(): ReduxState {
    return this.reduxStore.getState()
  }
}

if (window.Cypress) {
  window.store = Store.instance.reduxStore
}

export default Store.instance
