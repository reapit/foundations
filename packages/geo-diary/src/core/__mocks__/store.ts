export default {
  dispatch: jest.fn(),
  state: {
    auth: {
      loginSession: '1',
      desktopSession: '2',
    },
    online: true,
  },
}
