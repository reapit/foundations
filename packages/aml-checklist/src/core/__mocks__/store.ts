import { mockLoginSession } from '../../utils/__mocks__/session'

export default {
  dispatch: jest.fn(),
  state: {
    auth: {
      loginSession: mockLoginSession,
      refreshSession: null,
    },
  },
}
