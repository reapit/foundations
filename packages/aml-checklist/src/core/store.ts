import {
  createStore,
  applyMiddleware,
  compose,
  combineReducers,
  Store as ReduxStore,
  Dispatch,
  Reducer,
  CombinedState,
  AnyAction,
} from 'redux'
import { ReduxState, Action } from '../types/core'
import createSagaMiddleware from 'redux-saga'
import { all, fork } from '@redux-saga/core/effects'

import auth from '@/reducers/auth'
import checklistDetail from '@/reducers/checklist-detail'
import error from '@/reducers/error'
import result from '@/reducers/result'
import identityTypes from '@/reducers/identity-types'

import resultSagas from '@/sagas/result'
import authSagas from '@/sagas/auth'
import checklistDetailSagas from '@/sagas/checklist-detail'
import identityTypesSagas from '@/sagas/identity-types'
import { injectSwitchModeToWindow } from '@reapit/elements'

export class Store {
  static _instance: Store

  static get instance() {
    if (!Store._instance) {
      Store._instance = new Store()
    }

    return Store._instance
  }

  static isProd = process.env.NODE_ENV === 'production'

  static sagaMiddleware = createSagaMiddleware()

  static reducers = combineReducers({
    auth,
    error,
    result,
    checklistDetail,
    identityTypes,
  }) as Reducer<CombinedState<ReduxState>, Action<any> | AnyAction>

  static sagas = function*() {
    yield all([fork(authSagas), fork(checklistDetailSagas), fork(resultSagas), fork(identityTypesSagas)])
  }

  static composeEnhancers =
    !Store.isProd && window && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose

  reduxStore: ReduxStore<ReduxState>

  constructor() {
    injectSwitchModeToWindow()
    const composed = Store.composeEnhancers(applyMiddleware(Store.sagaMiddleware))

    this.reduxStore = createStore(Store.reducers, composed)

    Store.sagaMiddleware.run(Store.sagas)

    this.hotModuleReloading()
  }

  hotModuleReloading() {
    const hotModule = (module as any).hot

    if (hotModule) {
      // Enable Webpack hot module replacement for reducers
      hotModule.accept('../reducers', () => {
        this.reduxStore.replaceReducer(Store.reducers)
      })
    }
  }

  get dispatch(): Dispatch {
    return this.reduxStore.dispatch
  }

  get state(): ReduxState {
    return this.reduxStore.getState()
  }
}

export default Store.instance
