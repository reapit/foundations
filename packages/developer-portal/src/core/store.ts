import { createStore, applyMiddleware, compose, combineReducers, Store as ReduxStore, Dispatch } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { fork, all } from '@redux-saga/core/effects'
import { ReduxState } from '../types/core'
import appsReducer from '@/reducers/apps'
import scopesReducer from '@/reducers/scopes'
import categoriesReducer from '@/reducers/categories'
import desktopIntegrationTypesReducer from '@/reducers/desktop-integration-types'

import developer from '@/reducers/developer'
import error from '@/reducers/error'
import developerSetStatus from '@/reducers/developer-set-status'
import settingsReducer from '@/reducers/settings'
import appUsageStatsReducer from '@/reducers/app-usage-stats'
import trafficStatistics from '@/reducers/traffic-statistics'
import developerSubscriptions from '@/reducers/developer-subscriptions'
import developersReducer from '@/reducers/developers'
import installationsReducer from '@/reducers/installations'
import webhooksTopicsReducer from '@/reducers/webhooks-topics'
import webhooksSubscriptionsReducer from '@/reducers/webhooks-subscriptions'
import noticationMessage from '@/reducers/notification-message'

import {
  appDetailSagas,
  appListSagas,
  appAuthenticationSagas,
  createAppSagas,
  deleteAppSagas,
  appRevisionDetailSagas,
  appRevisionListlSagas,
  declineAppRevisionSagas,
  createAppRevisionSagas,
} from '@/sagas/apps'
import { scopeListSagas } from '@/sagas/scopes'
import { categoryListSagas } from '@/sagas/categories'
import { desktopIntegrationTypeListSagas } from '@/sagas/desktop-integration-types'
import appUsageStatsSagas from '@/sagas/app-usage-stats'
import { trafficStatisticsSagas } from '@/sagas/traffic-statistics'
import developerSagas from '@/sagas/developer'
import developerSetStatusSagas from '@/sagas/developer-set-status'
import settingSagas from '@/sagas/settings'
import { webhooksSubscriptionsSagas, webhooksEditSubscription } from '@/sagas/webhooks-subscriptions'
import { injectSwitchModeToWindow } from '@reapit/elements'
import developerSubscriptionsSagas from '@/sagas/developer-subscriptions'
import developersSagas from '@/sagas/developers'
import installationsSagas from '@/sagas/installations'
import { webhooksTopicsSagas } from '@/sagas/webhooks-topics'

export class Store {
  static _instance: Store

  static get instance() {
    if (!Store._instance) {
      Store._instance = new Store()
    }

    return Store._instance
  }

  static isProd = process.env.NODE_ENV === 'production'

  static sagaMiddleware = createSagaMiddleware()

  static reducers = combineReducers<ReduxState>({
    apps: appsReducer,
    scopes: scopesReducer,
    categories: categoriesReducer,
    desktopIntegrationTypes: desktopIntegrationTypesReducer,
    developer,
    error,
    developerSetStatus,
    settings: settingsReducer,
    appUsageStats: appUsageStatsReducer,
    noticationMessage,
    trafficStatistics,
    developerSubscriptions,
    developers: developersReducer,
    installations: installationsReducer,
    webhooksTopics: webhooksTopicsReducer,
    webhooksSubscriptions: webhooksSubscriptionsReducer,
  })

  static sagas = function*() {
    yield all([
      fork(appUsageStatsSagas),
      fork(developerSagas),
      fork(appDetailSagas),
      fork(appListSagas),
      fork(scopeListSagas),
      fork(categoryListSagas),
      fork(desktopIntegrationTypeListSagas),
      fork(appAuthenticationSagas),
      fork(createAppSagas),
      fork(deleteAppSagas),
      fork(appRevisionDetailSagas),
      fork(appRevisionListlSagas),
      fork(declineAppRevisionSagas),
      fork(createAppRevisionSagas),
      fork(developerSetStatusSagas),
      fork(settingSagas),
      fork(trafficStatisticsSagas),
      fork(webhooksEditSubscription),
      fork(developerSubscriptionsSagas),
      fork(developersSagas),
      fork(installationsSagas),
      fork(webhooksTopicsSagas),
      fork(webhooksSubscriptionsSagas),
    ])
  }

  static composeEnhancers =
    !Store.isProd && window && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose

  reduxStore: ReduxStore<ReduxState>

  constructor() {
    injectSwitchModeToWindow()
    const composed = Store.composeEnhancers(applyMiddleware(Store.sagaMiddleware))

    this.reduxStore = createStore(Store.reducers, composed)

    Store.sagaMiddleware.run(Store.sagas)

    this.hotModuleReloading()
  }

  hotModuleReloading() {
    const hotModule = (module as any).hot

    if (hotModule) {
      // Enable Webpack hot module replacement for reducers
      hotModule.accept('../reducers', () => {
        this.reduxStore.replaceReducer(Store.reducers)
      })
    }
  }

  get dispatch(): Dispatch {
    return this.reduxStore.dispatch
  }

  get state(): ReduxState {
    return this.reduxStore.getState()
  }
}

if (window.Cypress) {
  window.store = Store.instance.reduxStore
}

export default Store.instance
