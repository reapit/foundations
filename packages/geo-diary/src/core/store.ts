import {
  createStore,
  applyMiddleware,
  compose,
  combineReducers,
  Store as ReduxStore,
  Dispatch,
  AnyAction,
  CombinedState,
  Reducer,
} from 'redux'
import createSagaMiddleware from 'redux-saga'
import { ReduxState, Action } from '@/types/core'
import { all, fork } from '@redux-saga/core/effects'
import ActionTypes from '@/constants/action-types'

import home from '@/reducers/home'
import appointments from '@/reducers/appointments'
import appointmentDetail from '@/reducers/appointment-detail'
import nextAppointment from '@/reducers/next-appointment'
import error from '@/reducers/error'
import direction from '@/reducers/direction'
import auth from '@/reducers/auth'
import online from '@/reducers/online'

import authSagas from '@/sagas/auth'
import homeSagas from '@/sagas/home'
import appointmentsSagas from '@/sagas/appointments'
import appointmentDetailSagas from '@/sagas/appointment-detail'
import nextAppointmentSaga from '@/sagas/next-appointment'
import { injectSwitchModeToWindow } from '@reapit/elements'

export class Store {
  static _instance: Store

  static get instance() {
    if (!Store._instance) {
      Store._instance = new Store()
    }

    return Store._instance
  }

  static isProd = process.env.NODE_ENV === 'production'

  static sagaMiddleware = createSagaMiddleware()

  static reducers = combineReducers({
    online,
    auth,
    error,
    home,
    direction,
    appointments,
    appointmentDetail,
    nextAppointment,
  }) as Reducer<CombinedState<ReduxState>, Action<any> | AnyAction>

  static sagas = function*() {
    yield all([
      fork(authSagas),
      fork(homeSagas),
      fork(appointmentsSagas),
      fork(appointmentDetailSagas),
      fork(nextAppointmentSaga),
    ])
  }

  static composeEnhancers =
    !Store.isProd && window && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose

  reduxStore: ReduxStore<ReduxState>

  constructor() {
    injectSwitchModeToWindow()
    const composed = Store.composeEnhancers(applyMiddleware(Store.sagaMiddleware))
    this.reduxStore = createStore(Store.reducers, composed)

    Store.sagaMiddleware.run(Store.sagas)

    this.hotModuleReloading()

    // online/offline listeners
    window.addEventListener('online', () => this.dispatch({ type: ActionTypes.ONLINE }))
    window.addEventListener('offline', () => this.dispatch({ type: ActionTypes.OFFLINE }))
  }

  hotModuleReloading() {
    const hotModule = (module as any).hot

    if (hotModule) {
      // Enable Webpack hot module replacement for reducers
      hotModule.accept('../reducers', () => {
        this.reduxStore.replaceReducer(Store.reducers)
      })
    }
  }

  get dispatch(): Dispatch {
    return this.reduxStore.dispatch
  }

  get state(): ReduxState {
    return this.reduxStore.getState()
  }
}

export default Store.instance
