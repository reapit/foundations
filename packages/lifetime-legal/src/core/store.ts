import { createStore, applyMiddleware, compose, combineReducers, Store as ReduxStore, Dispatch } from 'redux'
import success from '@/reducers/success'
import auth from '@/reducers/auth'
import error from '@/reducers/error'
import results from '@/reducers/results'
import checklistDetail from '@/reducers/checklist-detail'
import identityTypes from '@/reducers/identity-types'

import { ReduxState } from '@/types/core'
import authSagas from '@/sagas/auth'
import successSagas from '@/sagas/success'
import identityTypesSagas from '@/sagas/identity-types'
import resultsSagas from '@/sagas/results'
import checklistDetailSagas from '@/sagas/checklist-detail'

import submitChecks from '../reducers/submit-checks'
import createSagaMiddleware from 'redux-saga'
import submitChecksSagas from '../sagas/submit-checks'
import { all, fork } from '@redux-saga/core/effects'
import { injectSwitchModeToWindow } from '@reapit/elements'

export class Store {
  static _instance: Store

  static get instance() {
    if (!Store._instance) {
      Store._instance = new Store()
    }

    return Store._instance
  }

  static isProd = process.env.NODE_ENV === 'production'

  static sagaMiddleware = createSagaMiddleware()

  static reducers = combineReducers({
    error,
    success,
    auth,
    results,
    submitChecks,
    checklistDetail,
    identityTypes,
  })

  static sagas = function*() {
    yield all([
      fork(resultsSagas),
      fork(authSagas),
      fork(successSagas),
      fork(submitChecksSagas),
      fork(checklistDetailSagas),
      fork(identityTypesSagas),
    ])
  }

  static composeEnhancers =
    !Store.isProd && window && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose

  reduxStore: ReduxStore<ReduxState>

  constructor() {
    injectSwitchModeToWindow()
    const composed = Store.composeEnhancers(applyMiddleware(Store.sagaMiddleware))

    this.reduxStore = createStore(Store.reducers, composed)

    Store.sagaMiddleware.run(Store.sagas)

    this.hotModuleReloading()
  }

  hotModuleReloading() {
    const hotModule = (module as any).hot

    if (hotModule) {
      // Enable Webpack hot module replacement for reducers
      hotModule.accept('../reducers', () => {
        this.reduxStore.replaceReducer(Store.reducers as any)
      })
    }
  }

  get dispatch(): Dispatch {
    return this.reduxStore.dispatch
  }

  get state(): ReduxState {
    return this.reduxStore.getState()
  }
}

export default Store.instance
