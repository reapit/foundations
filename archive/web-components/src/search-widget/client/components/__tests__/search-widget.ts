import SearchWidget from '../search-widget.svelte'
import { render } from '@testing-library/svelte'
import searchWidgetStore from '../../core/store'
import { get } from 'svelte/store'
import createGoogleMapsMock from '../../utils/__mocks__/mock-google-map'

const theme = {
  baseBackgroundColor: '',
  basefontSize: '',
  basefontColor: 'red',
  inverseFontColor: '',
  secondaryfontColor: '',
  primaryHeadingFontSize: '',
  secondaryHeadingFontSize: '',
  baseFontFamily: '',
  headingFontFamily: '',
  primaryAccentColor: '',
  secondaryAccentColor: '',
  mapAccentColor: '',
  breakPoints: {
    mobile: '',
    tablet: '',
    laptop: '',
    desktop: '',
  },
}

describe('search-widget', () => {
  beforeAll(() => {
    createGoogleMapsMock()
  })
  it('it matches a snapshot', () => {
    const wrapper = render(SearchWidget, {
      theme,
      apiKey: '',
      customerId: '',
      parentSelector: '#search-widget',
    })
    const { container } = wrapper

    expect(container).toMatchSnapshot()
  })

  it('it updates the default store on mount', () => {
    const props = {
      theme,
      apiKey: 'SOME_KEY',
      customerId: 'DEMO',
      parentSelector: '#search-widget',
    }
    render(SearchWidget, props)
    const store = get(searchWidgetStore)

    expect(store.initializers).toEqual(props)
  })
})
