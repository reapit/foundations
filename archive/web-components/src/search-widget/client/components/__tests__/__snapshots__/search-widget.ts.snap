// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-widget it matches a snapshot 1`] = `
.emotion-8 html,
.emotion-8 body,
.emotion-8 div,
.emotion-8 span,
.emotion-8 applet,
.emotion-8 object,
.emotion-8 iframe,
.emotion-8 h1,
.emotion-8 h2,
.emotion-8 h3,
.emotion-8 h4,
.emotion-8 h5,
.emotion-8 h6,
.emotion-8 p,
.emotion-8 blockquote,
.emotion-8 pre,
.emotion-8 a,
.emotion-8 abbr,
.emotion-8 acronym,
.emotion-8 address,
.emotion-8 big,
.emotion-8 cite,
.emotion-8 code,
.emotion-8 del,
.emotion-8 dfn,
.emotion-8 em,
.emotion-8 img,
.emotion-8 ins,
.emotion-8 kbd,
.emotion-8 q,
.emotion-8 s,
.emotion-8 samp,
.emotion-8 small,
.emotion-8 strike,
.emotion-8 strong,
.emotion-8 sub,
.emotion-8 sup,
.emotion-8 tt,
.emotion-8 var,
.emotion-8 b,
.emotion-8 u,
.emotion-8 i,
.emotion-8 center,
.emotion-8 dl,
.emotion-8 dt,
.emotion-8 dd,
.emotion-8 menu,
.emotion-8 ol,
.emotion-8 ul,
.emotion-8 li,
.emotion-8 fieldset,
.emotion-8 form,
.emotion-8 label,
.emotion-8 legend,
.emotion-8 table,
.emotion-8 caption,
.emotion-8 tbody,
.emotion-8 tfoot,
.emotion-8 thead,
.emotion-8 tr,
.emotion-8 th,
.emotion-8 td,
.emotion-8 article,
.emotion-8 aside,
.emotion-8 canvas,
.emotion-8 details,
.emotion-8 embed,
.emotion-8 figure,
.emotion-8 figcaption,
.emotion-8 footer,
.emotion-8 header,
.emotion-8 hgroup,
.emotion-8 main,
.emotion-8 menu,
.emotion-8 nav,
.emotion-8 output,
.emotion-8 ruby,
.emotion-8 section,
.emotion-8 summary,
.emotion-8 time,
.emotion-8 mark,
.emotion-8 audio,
.emotion-8 video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-8 article,
.emotion-8 aside,
.emotion-8 details,
.emotion-8 figcaption,
.emotion-8 figure,
.emotion-8 footer,
.emotion-8 header,
.emotion-8 hgroup,
.emotion-8 main,
.emotion-8 menu,
.emotion-8 nav,
.emotion-8 section {
  display: block;
}

.emotion-8 *[hidden] {
  display: none;
}

.emotion-8 body {
  line-height: 1;
}

.emotion-8 menu,
.emotion-8 ol,
.emotion-8 ul {
  list-style: none;
}

.emotion-8 blockquote,
.emotion-8 q {
  quotes: none;
}

.emotion-8 blockquote:before,
.emotion-8 blockquote:after,
.emotion-8 q:before,
.emotion-8 q:after {
  content: '';
  content: none;
}

.emotion-8 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.emotion-7 {
  font-size: 16px;
  font-family: Helvetica,Arial,sans-serif;
  background: #fff;
  color: red;
}

#search-widget .emotion-2 {
  font-size: 16px;
  font-family: Helvetica,Arial,sans-serif;
  margin-bottom: 1em;
}

#search-widget .emotion-1 {
  font-family: Helvetica,Arial,sans-serif;
  font-size: 18px;
  color: #000;
  border: 1px solid grey;
  background: #fff;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

#search-widget .emotion-1:hover {
  background: grey;
  color: #fff;
}

#search-widget .emotion-0 {
  font-family: Helvetica,Arial,sans-serif;
  font-size: 18px;
  border: 1px solid grey;
}

.emotion-5,
.emotion-5 svg {
  color: red;
}

#search-widget .emotion-6 {
  width: 100%;
}

@media screen and (min-width:960px) {
  #search-widget .emotion-6 {
    width: 33.333333%;
  }
}

@media screen and (min-width:560px) and (max-width:960px) {
  #search-widget .emotion-6 {
    width: 50%;
  }
}

<body>
  <div>
    <div
      class="emotion-8 emotion-7 search-widget svelte-gxdr06"
    >
      <form>
        <div
          class="search-form emotion-6 svelte-1iepkj4"
        >
          <input
            class="emotion-0 search-form-input svelte-1iepkj4"
            data-testid="search-form-input"
            id="search"
            type="text"
          />
           
           
          <button
            class="emotion-1 search-form-button svelte-1iepkj4"
            data-testid="btnSearch"
            type="submit"
          >
            Search Properties
          </button>
           
          <div
            class="search-form-control svelte-1iepkj4"
          >
            <label
              class="emotion-2 search-form-control-radio svelte-1iepkj4"
            >
              <input
                class="svelte-1iepkj4"
                data-testid="sales"
                name="searchType"
                type="radio"
                value="Sale"
              />
              
        For Sale
            </label>
             
            <label
              class="emotion-2 search-form-control-radio svelte-1iepkj4"
            >
              <input
                class="svelte-1iepkj4"
                data-testid="lettings"
                name="searchType"
                type="radio"
                value="Rent"
              />
              
        To Rent
            </label>
             
            <button
              class="emotion-2
        emotion-5 search-form-advanced-search svelte-1iepkj4"
              data-testid="btnAdvancedSearch"
              type="button"
            >
              Advanced Search
        
              <svg
                aria-hidden="true"
                class=""
                id=""
                role="img"
                style="height:1em;vertical-align:-.125em;overflow:visible;"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  transform="translate(256 256)"
                >
                  <g
                    transform=""
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                      transform="translate(-256 -256)"
                    />
                  </g>
                </g>
              </svg>
              
            </button>
          </div>
        </div>
         
      </form>
       
      <div
        class="search-widget-items-container svelte-gxdr06"
      >
        <div
          class="search-widget-items svelte-gxdr06"
        >
           
           
           
        </div>
         
        <div
          class="google-map-outer-container 
  emotion-7 svelte-ltkgd5"
        >
           
          <div
            class="google-map-container  svelte-ltkgd5"
          >
             
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
